src/Common/src/System/IO/PersistedFiles.Unix.cs

```diff
@@ -94,5 +94,95 @@ private static void EnsureUserDirectories()
                 TopLevelHiddenDirectory,
                 SecondLevelDirectory);
         }
+
+        /// <summary>Gets the current user's home directory.</summary>
+        /// <returns>The path to the home directory, or null if it could not be determined.</returns>
+        internal static string GetHomeDirectory()
+        {
+            // First try to get the user's home directory from the HOME environment variable.
+            // This should work in most cases.
+            string userHomeDirectory = Environment.GetEnvironmentVariable("HOME");
+            if (!string.IsNullOrEmpty(userHomeDirectory))
+                return userHomeDirectory;
+
+            // In initialization conditions, however, the "HOME" enviroment variable may 
+            // not yet be set. For such cases, consult with the password entry.
+            unsafe
```

Comment:
I went back and forth on this.  It's already cached in the places that use it: EnsureUserDirectories caches it as part of the larger string, and the only other consumer is Console which uses it to find the terminfo database, and it only does that once.  I was also slightly concerned about the home directory changing, either due to the environment variable getting changed or due to the a temporary change of user identity (e.g. seteuid)... though if we were actually concerned about that, we'd want to reevaluate the caching in EnsureUserDirectory as well, since it includes the home directory.  In any event, I think it makes sense to leave it for now without the caching, and then add it later if we determine it's important.  Ok?